app/components/Settings/Scheduler/previewModal.jsx

```diff
@@ -0,0 +1,248 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import { Controlled as CodeMirror } from 'react-codemirror2';
+import ms from 'ms';
+
+import Modal from '../../modal.jsx';
+import { Link } from '../../Link.react.js';
+import { FrequencySelector } from './createModal.jsx';
+import { Row, Column } from '../../layout.jsx';
+import SQL from './sql.jsx';
+import { plotlyUrl } from '../../../utils/utils.js';
+import { mapDialect } from './util.js';
+
+const NO_OP = () => {};
+
+const rowStyle = {
+    justifyContent: 'flex-start',
+    borderBottom: '1px solid rgba(0, 0, 0, 0.05)',
+    padding: '16px 0px'
+};
+const boxStyle = { boxSizing: 'border-box', width: '50%' };
+
+export class PreviewModal extends Component {
+    static defaultProps = {
+        onSave: NO_OP,
+        onDelete: NO_OP
+    };
+    static propTypes = {
+        query: PropTypes.object,
+        onSave: PropTypes.func,
+        onDelete: PropTypes.func,
+        onClickAway: PropTypes.func,
+        dialect: PropTypes.string
+    };
+    constructor(props) {
+        super(props);
+        this.state = {
+            editing: false,
+            code: props.query && props.query.query,
+            refreshInterval: props.query && props.query.refreshInterval,
+            confirmedDelete: false
+        };
+        this.toggleEditing = this.toggleEditing.bind(this);
+        this.updateCode = this.updateCode.bind(this);
+        this.updateRefreshInterval = this.updateRefreshInterval.bind(this);
+        this.onSubmit = this.onSubmit.bind(this);
+        this.onDelete = this.onDelete.bind(this);
+        this.close = this.close.bind(this);
+    }
+
+    componentDidUpdate(prevProps) {
```

Comment:
use `componentWillReceiveProps` instead

Reply:
Are you sure you want to use `componentWillReceiveProps`? That API is deprecated in `react@^16.3.0`. No worries either way, just wanted to make sure you knew 😄